src/api/api.ts

```diff
@@ -1,26 +1,76 @@
 import * as fs from 'fs';
-import fetch from 'node-fetch';
+import fetch, { Request, RequestInit } from 'node-fetch';
 import FormData from 'form-data';
 import {
   ConfigurationResponse,
   ClientConfig,
   AuthResponse,
   ClientAuth,
-  UploadThemeResponse,
+  ThemeResponse,
+  InitResponse,
+  Theme,
+  RoutesResponse,
+  ContentResponse,
+  GhostResponse,
 } from './api.interface';
 import { formatAuthParams } from './api.utils';
 import { Config } from './config';
+import { debugLog } from './debug-log';
+import getenv from 'getenv';
 
 export class ErrorFetchClientConfig extends Error {}
 export class ErrorAuthClient extends Error {}
 export class ErrorUploadTheme extends Error {}
+export class ErrorUploadContent extends Error {}
+export class ErrorUploadRoutes extends Error {}
+export class ErrorActivateTheme extends Error {}
+export class GhostAdminError extends Error {}
+export class RetryAdminRoute extends Error {
+  constructor(message?: string) {
+    super(message);
+
+    Object.setPrototypeOf(this, RetryAdminRoute.prototype);
+  }
+}
+
+export const RETRY_ERRORS = ['MaintenanceError'];
 
 export class GhostApi {
   private token: string = '';
 
   constructor(private config: Config) {}
 
-  public async init(): Promise<void> {
+  public async init(): Promise<InitResponse> {
+    const body = {
+      setup: [
+        {
+          name: 'ghost-upload-theme-bot',
+          email: this.config.credentials.email,
+          password: this.config.credentials.password,
+          blogTitle: 'Temporary Blog Title',
```

Comment:
Shouldn't `name` and `blogTitle` be into the config?

Reply:
Not really - it is immediately overwritten when you import the content